athena-jdbc/src/main/java/com/amazonaws/connectors/athena/jdbc/saphana/SAPHANARecordHandler.java

```diff
@@ -0,0 +1,94 @@
+import com.amazonaws.services.athena.AmazonAthena;
+import com.amazonaws.services.athena.AmazonAthenaClientBuilder;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import com.amazonaws.services.secretsmanager.AWSSecretsManagerClientBuilder;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.apache.commons.lang3.Validate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+/**
+ * Data handler, user must have necessary permissions to read from necessary tables.
+ */
+public class SAPHANARecordHandler
+        extends JdbcRecordHandler
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(SAPHANARecordHandler.class);
+
+    private static final String SAPHANA_QUOTE_CHARACTER = "\"";
+
+    private final JdbcSplitQueryBuilder jdbcSplitQueryBuilder;
+
+    /**
+     * Instantiates handler to be used by Lambda function directly.
+     *
+     * Recommend using {@link com.amazonaws.connectors.athena.jdbc.MultiplexingJdbcCompositeHandler} instead.
+     */
+    public SAPHANARecordHandler()
+    {
+        this(JDBCUtil.getSingleDatabaseConfigFromEnv(JdbcConnectionFactory.DatabaseEngine.SAPHANA));
+    }
+
+    public SAPHANARecordHandler(final DatabaseConnectionConfig databaseConnectionConfig)
+    {
+        this(databaseConnectionConfig, AmazonS3ClientBuilder.defaultClient(), AWSSecretsManagerClientBuilder.defaultClient(), AmazonAthenaClientBuilder.defaultClient(),
+                new GenericJdbcConnectionFactory(databaseConnectionConfig, SAPHANAMetadataHandler.JDBC_PROPERTIES), new SAPHANAQueryStringBuilder(SAPHANA_QUOTE_CHARACTER));
+    }
+
+    @VisibleForTesting
+    SAPHANARecordHandler(final DatabaseConnectionConfig databaseConnectionConfig, final AmazonS3 amazonS3, final AWSSecretsManager secretsManager,
+            final AmazonAthena athena, final JdbcConnectionFactory jdbcConnectionFactory, final JdbcSplitQueryBuilder jdbcSplitQueryBuilder)
+    {
+        super(amazonS3, secretsManager, athena, databaseConnectionConfig, jdbcConnectionFactory);
+        this.jdbcSplitQueryBuilder = Validate.notNull(jdbcSplitQueryBuilder, "query builder must not be null");
+    }
+
+    @Override
+    public PreparedStatement buildSplitSql(Connection jdbcConnection, String catalogName, TableName tableName, Schema schema, Constraints constraints, Split split)
+            throws SQLException
+    {
+        PreparedStatement preparedStatement = jdbcSplitQueryBuilder.buildSql(jdbcConnection, null, tableName.getSchemaName().toUpperCase(), tableName.getTableName().toUpperCase(), schema, constraints, split);
+
+        // Disable fetching all rows.
+        //preparedStatement.setFetchSize(Integer.MIN_VALUE);
```

Comment:
why is this commented?

Reply:
Removed the statement as MIN_VALUE is -ve which breaks JDBC driver. However LIMIT works fine